src/app/components/change-email/change-email.component.ts

```diff
@@ -0,0 +1,116 @@
+import { ConfirmationDialogComponent } from './../confirmation-dialog/confirmation-dialog.component';
+import { ChangeEmailService } from './../../services/change-email.service';
+import { Component, OnInit } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { FormGroup, FormControl, Validators, FormBuilder } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatDialog } from '@angular/material';
+import { MatSnackBar } from '@angular/material';
+import { AuthService } from './../../services/auth.service';
+
+@Component({
+  selector: 'app-change-email',
+  templateUrl: './change-email.component.html',
+  styleUrls: ['./change-email.component.css']
+})
+export class ChangeEmailComponent implements OnInit {
+  ngUnsubscribe = new Subject();
+  createEmailForm: FormGroup;
+  submitted: boolean;
+  $currentUser: Subscription;
+  static PATH: any = 'settings/email';
+
+  userInfo = '';
+  userId = '';
+
+  constructor(private changeEmailService: ChangeEmailService, private formBuilder: FormBuilder,
+    private route: ActivatedRoute, private router: Router, public dialog: MatDialog,
+    private snackBar: MatSnackBar, private auth: AuthService) {}
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(params => {
+      let newEmail = '';
+      let password = '';
+
+      this.changeEmailService.getCurrentUser().subscribe(
```

Comment:
You shouldn't be calling `getCurrentUser()` multiple times as it uses unnecessary bandwidth

Reply:
Changed it.